net/tstun/table/table.go

```diff
@@ -73,7 +58,7 @@ func (t *RoutingTableBuilder) Build() *RoutingTable {
 //
 // It is safe for concurrent use.
 type RoutingTable struct {
-	prefixTrie *device.AllowedIPs
+	table *bart.Table[key.NodePublic]
```

Comment:
the caller really just wants the addresses but we had to go through the `key.NodePublic` abstraction previously. See my other PR, we should really just have `bart.Table[netip.Addr]` here

Reply:
or in the layer above, we could just delete this package

Reply:
The thing that this abstraction adds is the login in `InsertOrReplace` that handles changed prefixes for a given peer by first deleting the old ones before adding the new ones. I noticed that your PR dropped that and I wasn't sure that was safe to do, as I don't know what exactly depends on that behavior but I assume it was done for a reason.

Reply:
The whole table is atomically replaced in tstun. We don't use the replace logic at all

Reply:
Coo, so `wcfg.Peers` is guaranteed not to contain duplicates?

Reply:
Yep

Reply:
Done